cmd/kubeadm/app/constants/constants.go

```diff
@@ -177,3 +180,17 @@ func GetAdminKubeConfigPath() string {
 func AddSelfHostedPrefix(componentName string) string {
 	return fmt.Sprintf("%s%s", SelfHostingPrefix, componentName)
 }
+
+// CreateTempDirForKubeadm is a function that creates a temporary directory under /etc/kubernetes/tmp (not using /tmp as that would potentially be dangerous)
+func CreateTempDirForKubeadm(dirName string) (string, error) {
+	// creates target folder if not already exists
+	if err := os.MkdirAll(TempDirForKubeadm, 0700); err != nil {
```

Comment:
root only on TempDir?

Reply:
yes; we concluded that there are possible, pretty nasty `/tmp`-dir attacks where lots of users can do things; so we moved the upgrade temp dir to `/etc/kubernetes/tmp` with root-only access